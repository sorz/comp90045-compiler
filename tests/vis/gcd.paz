program gcd;

{ Greatest common divisor of two positive integers
  using a variant of Euclid's method }

var
    x, y, temp, quotient, remainder: integer;

begin
    write('Input a positive integer: ');
    read(x);
    write('And another positive integer: ');
    read(y);
    writeln;

    if x < y then
    begin
        temp := x;
        x := y;
        y := temp
    end;

    write('The gcd of ');
    write(x);
    write(' and ');
    write(y);
    write(' is ');

    quotient := x div y;
    remainder := x - quotient * y;

    while remainder > 0 do
    begin
        x := y;
        y := remainder;
        quotient := x div y;
        remainder := x - quotient * y
    end;

    write(y);
    writeln
end.
