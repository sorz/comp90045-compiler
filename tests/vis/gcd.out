Input a positive integer: And another positive integer: 
The gcd of 21 and 14 is 7
